azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+
+        # get file extension
+        file_name = os.path.basename(file)
+        cert_extension = validate_file_extension(file_name, [".der", ".crt", ".crl"])
+        # get cert name by removing extension
+        cert_name = file_name.replace(f"{cert_extension}", "")
+
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        if cert_extension == ".crl":
+            if opcua_secret_sync:
+                secret_mapping = opcua_secret_sync.get("properties", {}).get("objectSecretMapping", [])
+                possible_file_names = [f"{cert_name}.crt", f"{cert_name}.der"]
+                matched_names = [
+                    mapping["targetKey"] for mapping in secret_mapping if mapping["targetKey"] in possible_file_names
+                ]
+
+            if not opcua_secret_sync or not matched_names:
+                logger.error(f"Cannot add .crl {file_name} without corresponding .crt or .der file.")
+                return
+
+        secret_name = secret_name if secret_name else file_name.replace(".", "-")
+
+        # iterate over secrets to check if secret with same name exists
+        secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+        self._upload_to_key_vault(secret_name, file, cert_extension)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        self._add_secrets_to_spc(
+            secrets=[secret_name],
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+            spc_tenant_id=spc_tenant_id,
+            spc_client_id=spc_client_id,
+        )
+
+        return self._add_secrets_to_secret_sync(
+            secrets=[(secret_name, file_name)],
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+        )
+
+    def client_add(
+        self,
+        instance_name: str,
+        resource_group: str,
+        public_key_file: str,
+        private_key_file: str,
+        subject_name: str,
+        application_uri: str,
+    ) -> dict:
+        # inform user if the provided cert was issued by a CA, the CA cert must be added to the issuers list.
+        logger.warning("Please ensure the certificate must be added to the issuers list if it was issued by a CA. ")
```

Comment:
will you check for the user?

Reply:
not for now, this is post GA nice to have, it will be pretty complicated to do and we don't have enough time